### ISSUE TITLE
Upper part of windows don't follow color scheme in some circumstances

### ISSUE THREAD

**Issue by stefonarch** [lxqt/lxqt#2192]
When changing widget style and palettes this can happen:

![schermata-02-16-09-13](https://user-images.githubusercontent.com/10681413/154224536-90db7ead-537a-4a40-9a55-e3c19aacf26c.png)

##### Steps to Reproduce (for bugs)

1. Load dark palette and icons for dark theme, fusion widget style
2. reset palette
3. switch to another widget style
4. reload dark palette

It's hard to produce - fusion, qtcurve and breeze will produce this artefact and it's also  hard to get back to normal. Switching widget styles to breeze, qtcurve and fusion will make impossible to change palette:

![schermata-02-16-09-45](https://user-images.githubusercontent.com/10681413/154228033-509a840e-25a0-4636-97ee-e860fdaac361.png)

##### Context

Working on a theme at some point I switched to breeze for transparency, and today I found this artefact.

##### System Information

* Distribution & Version: git

**Comment by palinek** [comment]
Changing widget style requires application restart. Is the reported issue visible after re-launching the app?

**Comment by stefonarch** [comment]
I tried even rebooting... I tried in VM Manjaro, same. Maybe it's necessary not to hit "apply" change palette and widget style together then apply. Actually I didn't find any way back to normal now, once I did. Will test later.

**Comment by stefonarch** [comment]
I see that restarting apps or session solves in VWwith 1.0.0  but not in my system. 
Have to test my testuser in the system more.

**Comment by tsujan** [comment]
Restyling Qt apps *on the fly* is always problematic. It can be done more or less but issues may be found here and there. Restarting apps or the session by logging out and in is recommended (no reboot is needed).

I'm not sure about the cause of your problem but Breeze doesn't take all of its colors from Qt palette. It has internal colors that can be set only by changing KDE's "color scheme". QtCurve can also have internal colors. Kvantum completely ignores the Qt palette that's set by the DE. Only Fusion is guaranteed to follow Qt palette completely.

**Comment by tsujan** [comment]
Also, make sure `qt5ct` isn't active.

**Comment by stefonarch** [comment]
The issue is that restart desktop module or lximage-qt or session doesn't fix the appearance.
I switched session to the test user and the artifact was still there from start, so it's in my system.
Differences to the Manjaro VM with 1.0.0 are git and wipqt6 installed, may be others. Will try on my desktop PC later (only git).

**Comment by tsujan** [comment]
If it really isn't changed after app restart, Breeze may have got a new bug (or "feature").

Anyway, I'm almost sure that it's out of our control how a Qt style behaves when the style changes to it.

**Comment by stefonarch** [comment]
I've still  no clue, on my main PC  with latest git+arch (which is the "family" PC and doesn't suffer all my tests) all changes are perfectly applied on the fly,  I could not reproduce even one of those artefacts.

**Comment by stefonarch** [comment]
Looks as if I nailed it down. 
Testing again a freshly created user the issue is not present, so it had to be some configuration file, and in fact deleting `~/.config/kdeglobals` everything turns back to normal, changes of widget style even are applied on the fly,

**Comment by tsujan** [comment]
I'm not saying that I understand what has happened but I remembered that, a few times, after upgrades, the style (and even the icon theme) couldn't be changed cleanly under KDE — not even after logging out and in. I solved such problems by removing KDE/Plasma cache files from `~/.cache`.

On one occasion, Dolphin took a lot of CPU time to start. The removal of KDE cache files didn't help but the removal of `~/.local/share/dolphin` did.

I guess what you encountered was about a similar KDE bug.
